### DISCUSSION TITLE
ISO install results in / with 50 GB partition, /nsm not assigned to its own drive

### DISCUSSION THREAD

**Discussion by hershmaster** [Security-Onion-Solutions/securityonion#1791] [Unsupported Versions] [answered]
Security Onion Version 2.3.2
OS: CentOS 7
Source: ISO
Architecture: Standalone

sda is 1TB, and was selected for main OS install.  sdb is 2TB, and was selected for NSM.

On my first install, I selected LVM for disk partitioning.  After the installation was complete, I checked the drives and found that /nsm was neither formatted or mounted from /dev/sdb.  Also, / was assigned a 50 GB partition that was quickly filling up due to /nsm being under it.  Here's the "df -h" output from that install after I manually set up /nsm on /dev/sdb (thanks to the commands provided in #1767 by @TOoSmOotH ).

![LVM](https://user-images.githubusercontent.com/1329169/97949405-61c3ba00-1d48-11eb-8c76-ae38e2e3d3bd.png)

On a subsequent install, I selected the standard option for partitioning.  After installation, I again saw that / had been assigned to a 50 GB partition, with /nsm underneath (and filling up).  Also, /home got assigned to all of /dev/sdb. Here's a screenshot of "df -h" output before I made any changes.

![standard](https://user-images.githubusercontent.com/1329169/97949483-99326680-1d48-11eb-8e8d-984a3006fad6.png)

I was able to get things moved around (again with the help of the commands in #1767) and currently have this build running now.  It seems to be stable for now.

My questions at this point are:

1. Did I miss something during the install that would have let me direct the partitioning manually?
2. Would you have any advice for best way to extend the / volume to take up the rest of the disk? (/ is on /dev/sda4, the last partition on the drive).

Thanks again.

-Steve H

**Comment by TOoSmOotH** [comment]
Let me try and recreate the issue.

**Comment by netadmn475** [comment]
I'm having the same problem. Using version 2.3.1 btw.

<img width="573" alt="Screen Shot 2020-11-05 at 4 49 49 AM" src="https://user-images.githubusercontent.com/74000373/98243111-588e4500-1f22-11eb-8735-d5dcd4923da9.png">

**Reply by dougburks** [reply]
Are you using our Security Onion ISO image or a CentOS ISO image? If CentOS, did you review the partitioning guidance in our documentation?
https://docs.securityonion.net/en/2.3/partitioning.html

**Reply by netadmn475** [reply]
Used Cent OS image and followed the partitioning guidance in the documentation.

**Reply by dougburks** [reply]
I'm not sure I understand. Your screenshot looks like default CentOS partitioning. In the CentOS installer, did you change the default CentOS partitioning per the documentation?

**Reply by netadmn475** [reply]
I'm trying to go back and reinstall and see if I missed something...

**Comment by TOoSmOotH** [comment]
So I was able to test this with 2.3.2 and I had now issues using 2 disks. 

Here is what my lsblk looks like:
```
NAME            MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sda               8:0    0  100G  0 disk
├─sda1            8:1    0    1M  0 part
├─sda2            8:2    0  500M  0 part /boot
└─sda3            8:3    0 99.5G  0 part
  └─system-root 253:0    0 99.5G  0 lvm  /
sdb               8:16   0 1000G  0 disk
└─sdb1            8:17   0 1000G  0 part
  └─nsm-nsm     253:1    0 1000G  0 lvm  /nsm
sr0              11:0    1  6.9G  0 rom
```

**Comment by hershmaster** [comment]
And this is mine after a fresh attempt:

    NAME            MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
    sda               8:0    0 931.5G  0 disk 
    ├─sda1            8:1    0   200M  0 part /boot/efi
    ├─sda2            8:2    0     1G  0 part /boot
    └─sda3            8:3    0 930.3G  0 part 
      ├─centos-root 253:0    0    50G  0 lvm  /
      ├─centos-swap 253:1    0  15.7G  0 lvm  [SWAP]
      └─centos-home 253:2    0   2.7T  0 lvm  /home
    sdb               8:16   0   1.8T  0 disk 
    └─sdb1            8:17   0   1.8T  0 part 
      └─centos-home 253:2    0   2.7T  0 lvm  /home
    sr0              11:0    1   6.9G  0 rom 

This time I took pics of the initial installation steps, and I noticed an error during the "Generating updated storage configuration" section:

    storage configuration failed: Failed to find a suitable stage1 device: EFI System Partition cannot be of type xfs.: EFI System Partition must be mounted on one of /boot/efi.; EFI System Partition cannot be of type lvmpv.; EFI System Partition cannot be of type biosboot.

The installer proceeded to go on, but when it got to "Installation Options", option 5 "Installation Destination (Custom partitioning selected)" an ! instead of an x. I then had to go in and configure that option. I selected:

* Option 3 -> Select all (to pick both drives)
* Option 2 -> Use all space
* Option 3 -> LVM

and proceeded with the rest of the installation.

Maybe I need to tweak one of those options?

![IMG_0686](https://user-images.githubusercontent.com/1329169/98330616-a43b0000-1faf-11eb-9309-1c77c5cd26a9.jpg)

**Comment by hershmaster** [comment, marked as answer]
Finally figure this one out. Ultimately it had to do with how I was booting the install media.  I've got an old desktop machine with a DVD drive, and I was using the ISO burned on to a DVD to boot the installer.  In the Boot Options menu I'd bring up to access the DVD drive, I had a few different sections with options to choose from:

![IMG_0690](https://user-images.githubusercontent.com/1329169/98472506-5d642a80-21a8-11eb-91a0-735f8736469d.jpg)

I'd been choosing the DVD option under the **UEFI BOOT** section, and that was screwing up the installer later on. When I chose the DVD option under **LEGACY BOOT**, the installer moved through its steps with no issues or hiccups, and formatted the disks as expected.

Side note: I later had to disable UEFI in my BIOS/UEFI so that my sda drive would be recognized as bootable and let the OS come up.

But, I'm up and running now. Time to tune my rules now.  Thanks again for your help.

**Reply by netadmn475** [reply]
I'm going to try this now.

**Reply by netadmn475** [reply]
Hmm I don't have a  UEFI with Legacy Boot option.
